include/winsparkle.h

```diff
@@ -242,6 +242,38 @@ WIN_SPARKLE_API void __cdecl win_sparkle_set_app_build_version(const wchar_t *bu
  */
 WIN_SPARKLE_API void __cdecl win_sparkle_set_registry_path(const char *path);
 
+/// Type used to override WinSparkle's config read function
+typedef int(__cdecl *win_sparkle_config_read_t)(void *custom_data, const char *name, wchar_t *buf, size_t bytes);
+/// Type used to override WinSparkle's config write function
+typedef int(__cdecl *win_sparkle_config_write_t)(void *custom_data, const char *name, const wchar_t *value);
+/// Type used to override WinSparkle's config delete function
+typedef int(__cdecl *win_sparkle_config_delete_t)(void *custom_data, const char *name);
+
+/**
+    Override WinSparkle's configuration read, write and delete functions.
+
+    By default, WinSparkle will read, write and delete configuration values by
+    interacting directly with Windows Registry.
+    If you want to manage configuration by yourself or you don't want let WinSparkle 
+    write settings directly to the Windows Registry, you can provide your own functions 
+    to read, write and delete configuration.
+
+    These functions needs to return TRUE on success, FALSE on failure.
+    If you passed NULL as a configuration action (read, write or delete)'s function pointer, 
+    WinSparkle will use the default function for that action.
+
+    @param custom_data  arbitrary data which will be passed to your config manipulating 
+                        functions, WinSparkle will not read or alter it.
+
+    @note The forth parameter *bytes* of function config_read is the size of *buf* in *bytes*, 
```

Comment:
this would confuce so many users
there is no reasons, why this shouldn't be an array length

Reply:
...Because the original WinSparkle `RegistryRead(name, buf, sizeof(buf))` use the number of bytes (as you can see from `sizeof(buf)`), and I've left it unchanged.
But I think you're right, I'll change the code so it uses array length.

Reply:
I don't really understand what you mean by "array length" and how it differs from "buffer size"? The code this is derived from uses buffer length as passed to win32 functions and is the standard way to declare how much space is available for writing...

Reply:
If we have a `wchar_t buf[512]`, the array length is 512, but the buffer size is 1024. And in existing codes WinSparkle pass 1024 instead of 512, so I decided to keep it unchanged and follow this convention.
But I agree it's pretty counterintuitive, I've added that "it's buffer size in bytes, not array length!" Exactly because I have experienced crashes due memory corruption...

Reply:
Imagine next example:
```cpp
int __cdecl config_read_impl(void *custom_data, const char *name, wchar_t *buf, size_t bytes)
{
    std::wstring the_value("This is some very long (or not) value - got from our reader"); // 60+1 character
    size_t max_len_to_write = std::max(the_value.size(), bytes-1);
    wcsncpy(buf, the_value.c_str(), max_len_to_write);
    buf[max_len_to_write] = L'\0';
    return TRUE;
}
```
and the fact, that `sizeof(wchar_t) == 2`
so many users will find a memory corruption here or broken values

despite the fact, that usage is not as per documentation, but this way is much more expected

Reply:
I've updated the code, now the function accepts array length instead of size in bytes.